### PR TITLE
[FLINK-23999][table-planner] Support evaluating individual window table-valued function in planner

### PR DESCRIPTION
**Pull request by beyond1920**
## What is the purpose of the change

This pr aims to support evaluating individual window table-valued function in planner

## Brief change log

  - Remove `SimplifyWindowTableFunctionRules`
  - Remove `emitPerRecord`field in `StreamPhysicalWindowTableFunction` and `StreamExecWindowTableFunction` because the behavior of window table function with row semantics is unified, they emit result by per record.
  - Refactor  `StreamExecWindowTableFunction` to support proctime window

## Verifying this change
  - UT in `WindowTableFunctionTest`
  - Harness in `WindowTableFunctionHarnessTest` to test process time window 
  - IT in `WindowTableFunctionITCase`
  - Json test in `WindowTableFunctionJsonPlanTest`

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (yes)
  - If yes, how is the feature documented? (docs)

### CODE REVIEW

File: flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecWindowTableFunction.java [hunk 1]
```diff
@@ -48,20 +42,15 @@
 public class StreamExecWindowTableFunction extends CommonExecWindowTableFunction
         implements StreamExecNode<RowData> {
 
-    public static final String FIELD_NAME_EMIT_PER_RECORD = "emitPerRecord";
-
-    @JsonProperty(FIELD_NAME_EMIT_PER_RECORD)
-    private final Boolean emitPerRecord;
+    public static final String FIELD_NAME_WINDOWING = "windowing";
```
Comment (godfreyhe) [inline flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecWindowTableFunction.java:45]: use the parent class's `FIELD_NAME_WINDOWING` ?

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/stream/sql/WindowTableFunctionTest.scala [hunk 2]
```diff
@@ -86,6 +87,17 @@ class WindowTableFunctionTest extends TableTestBase {
...
 
+  @Test
+  def testCumulateTVFProctime(): Unit = {
+    val sql =
+      """
+        |SELECT *
+        |FROM TABLE(
+        | CUMULATE(TABLE MyTable, DESCRIPTOR(proctime), INTERVAL '10' MINUTE, INTERVAL '1' HOUR))
+        |""".stripMargin
+    util.verifyExplain(sql)
```
Comment (godfreyhe) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/stream/sql/WindowTableFunctionTest.scala:98]: use `util.verifyExecPlan()`

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/rules/physical/stream/StreamPhysicalWindowTableFunctionRule.scala [hunk 3]
```diff
@@ -58,8 +58,7 @@ class StreamPhysicalWindowTableFunctionRule  extends ConverterRule(
       traitSet,
       newInput,
       scan.getRowType,
-      convertToWindowingStrategy(windowTableFunction, inputRowType),
-      false
+      convertToWindowingStrategy(scan.getCall.asInstanceOf[RexCall], newInput.getRowType)
```
Comment (godfreyhe) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/rules/physical/stream/StreamPhysicalWindowTableFunctionRule.scala:61]: revert the change ?

File: flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/physical/stream/SimplifyWindowTableFunctionRules.java [hunk 4]
```diff
@@ -1,348 +0,0 @@
...
-
-import java.util.Collections;
-import java.util.List;
-
-/**
- * Planner rule that tries to simplify emit behavior of {@link StreamPhysicalWindowTableFunction} to
- * emit per record instead of emit after watermark passed window end if time attribute is event-time
- * and followed by {@link StreamPhysicalWindowRank} or {@link StreamPhysicalWindowJoin}.
- */
-public interface SimplifyWindowTableFunctionRules {
```
Comment (godfreyhe) [inline flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/physical/stream/SimplifyWindowTableFunctionRules.java:41]: why no exec plans are change after this rule is deleted?

**Review by godfreyhe — COMMENTED** [review]
Thanks for the contribution @beyond1920 , I left some comments

File: flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/physical/stream/SimplifyWindowTableFunctionRules.java (hunk 4 above)
Comment (beyond1920) [reply flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/physical/stream/SimplifyWindowTableFunctionRules.java:41]: Those rules are used to attach a flag on window TVF which is followed after Window TopN/ Window Deduplicate/ Window Join. For those window TVF with flags would be normally executed in ExecNode, for other window TVF without flags would be thrown out in ExecNode.

Since we supported individual window table-valued function in planner, there is no need to distinguish window tvf based on the flags. So we removed those rules.

And there are some test which failed before could work now. For example, `WindowTableFunctionTest#testUnsupported`.

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/stream/sql/WindowTableFunctionTest.scala (hunk 2 above)
Comment (beyond1920) [reply flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/stream/sql/WindowTableFunctionTest.scala:98]: I original use util.verifyExecPlan() in order to cover the original `WindowTableFunctionTest#testUnsupported` could work fine now.
However, I think I could use `util.verifyExecPlan()` because `WindowTableFunctionITCase` already cover this.

**Comment by beyond1920** [comment]
@flinkbot run azure

**Review by godfreyhe — APPROVED** [review]
